### PR TITLE
more complete python gitignore

### PR DESCRIPTION
**Pull request by heartsucker**
Because currently jumping branches will leave junk like `.cache` floating around

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,40 +28,105 @@ Gemfile.lock
...
 .installed.cfg
-lib
-lib64
+*.egg
+
+# PyInstaller
+#  Usually these files are written by a python script from a template
+#  before PyInstaller builds the exe, so as to inject date/other infos into it.
+*.manifest
+*.spec
```
Comment (redshiftzero) [inline .gitignore]: No use of PyInstaller in SD, so I think this can be snipped out

File: .gitignore [hunk 2]
```diff
@@ -28,40 +28,105 @@ Gemfile.lock
...
+*,cover
+.hypothesis/
 
 # Translations
 *.mo
+*.pot
+
+# Django stuff:
+*.log
+local_settings.py
```
Comment (redshiftzero) [inline .gitignore]: No use of Django in SD, so I think this is safe to snip out too

File: .gitignore [hunk 3]
```diff
@@ -28,40 +28,105 @@ Gemfile.lock
...
+
+# Flask stuff:
+instance/
+.webassets-cache
+
+# Scrapy stuff:
+.scrapy
```
Comment (redshiftzero) [inline .gitignore]: Safe to snip this scrapy stuff out

File: .gitignore [hunk 4]
```diff
@@ -28,40 +28,105 @@ Gemfile.lock
...
+# dotenv
+.env
+
+# virtualenv
+.venv/
+venv/
+ENV/
+
+# Spyder project settings
+.spyderproject
```
Comment (redshiftzero) [inline .gitignore]: Snip Spyder too

File: .gitignore [hunk 5]
```diff
@@ -28,40 +28,105 @@ Gemfile.lock
...
+
+# Rope project settings
+.ropeproject
```
Comment (redshiftzero) [inline .gitignore]: And rope

File: .gitignore [hunk 6]
```diff
@@ -28,40 +28,105 @@ Gemfile.lock
...
+target/
+
+# Jupyter Notebook
+.ipynb_checkpoints
+
+# pyenv
+.python-version
+
+# celery beat schedule file
+celerybeat-schedule
```
Comment (redshiftzero) [inline .gitignore]: No use of Celery in SD so snip this out

File: .gitignore [hunk 7]
```diff
@@ -28,40 +28,105 @@ Gemfile.lock
...
+.scrapy
+
+# Sphinx documentation
+docs/_build/
+
+# PyBuilder
+target/
+
+# Jupyter Notebook
+.ipynb_checkpoints
```
Comment (redshiftzero) [inline .gitignore]: Probably another one to snip out and just have in your global gitignore

File: .gitignore [hunk 8]
```diff
@@ -28,40 +28,105 @@ Gemfile.lock
...
+.webassets-cache
+
+# Scrapy stuff:
+.scrapy
+
+# Sphinx documentation
+docs/_build/
+
+# PyBuilder
+target/
```
Comment (redshiftzero) [inline .gitignore]: No project-specific use of PyBuilder in SD so another one to snip out

**Comment by fowlslegs** [comment]
Not sure why the weird Travis failure in that `tearDown` function run. Anyway, using Github's "Update Branch," which will also re-run Travis.

**Comment by conorsch** [comment]
LGTM. @redshiftzero's in-line edit requests have been addressed by @heartsucker, so I'm going to mark as approved and merge.

**Pull request merged**